spring-integration-jms/src/test/java/org/springframework/integration/jms/MockBeanFactory.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2002-2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.jms;
+
+import org.springframework.beans.factory.BeanFactory;
+import org.springframework.integration.context.IntegrationContextUtils;
+import org.springframework.scheduling.TaskScheduler;
+import org.springframework.scheduling.concurrent.ThreadPoolTaskScheduler;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * @author Gengwu Zhao
+ *
+ * @since 6.3.1
+ *
+ */
+
+public class MockBeanFactory {
```

Comment:
This name is also wrong for public API.
And it also does not explain what it does to let other classes to use it.
I also wonder if we can use `TestUtils.createTestApplicationContext()` instead which has an expected bean registered.
Right, we might need to think about closing that ctx in the end of test, but that is indeed an existing API for the goal of your change: refactor code to minimize duplication.